fplll/bkz.cpp

```diff
@@ -499,6 +493,18 @@ template <class FT> bool BKZReduction<FT>::bkz()
   cputime_start = cputime();
 
   m.discoverAllRows();
+  
+  if (sld)
+  {
+    m.updateGSO();
+    sld_potential = m.get_slide_potential(0, num_rows, param.block_size);
+  }
+  
+  // the following is necessary, since sd-bkz starts with a dual tour and 
+  // svp_reduction calls size_reduction, which needs to be preceeded by a 
+  // call to lll lower blocks to avoid seg faults
+  if (sd)
+    lll_obj.lll(0, 0, num_rows);
```

Comment:
usually this is handled by the wrapper in the bkz_reduction function in fplll.cpp (except if the BKZ_NO_LLL flag is set, in which case calling lll beforehand would be incorrect). so the input is already lll reduced. the issue is just that certain variables are not initialized in the lll_obj that we have at hand in the bkz object, so size_reduction can segfault if called on lower blocks right away (as we do in sd, but not in bkz or sld).